### ISSUE TITLE
CBT Support in Citrix

### ISSUE THREAD

**Issue by mauzilla** [xenserver/xs-cbt-samples#18]
Hello,

Where are we supposed to ask for support for CBT related questions? I have 2 seperate open threads on the Citrix community forum but no one is answering. This is concerning as the introduction into CBT is a multi technology jump (using both CBT and NBD services) and it goes without saying that some developers may have more technical questions that what is covered in the developer guide.

For me the most important question right now is how does CBT work along with other backups? We would ideally like primarily rely on CBT incremental backups as our primary backup solution but still do a monthly full export of our servers (just incase CBT has some data corruption). Currently we take a snapshot and do a full export on a weekly basis, but this is very expensive method of doing things (time and storage). I assume that when we go CBT, we will not be able to make further snapshots of a VM without that snapshot VDI's being considered as another snapshot and having CBT data skewed?

Secondly, in the developer docs you refer to the possibility of comparing whether exported VDI matches that of the VDI snapshot via some kind of MD5 checksum. Any guide on how to achieve this? We would like to periodically ensure that what we have exported is correct and that no corruption took place without us knowing?

**Comment by olivierlambert** [comment]
Unrelated to CBT @mauzilla : why not using Xen Orchestra? You can do delta backup, and also choose to have one full at some point automatically to be sure you won't have any corruption in the chain after a while.

**Comment by mauzilla** [comment]
Hi Olivier,

It's a 2 fold problem, I tried installing from sources, got it to work but then ran into a snag and couldn't get it running again. I tried reinstalling but with extremely limited experience with the product (and knowing I am installing from sources == no support) I gave the CBT option a go. I am a developer myself so saw it as a learning curve which I was happy to take

We're paying Citrix extra money each month for "enterprise" licenses only because we need the CBT option. Unfortunately the difference in licensing does not cover the $550 route we would like to ultimately have (as someone needs to take ownership of the backups and if something goes wrong, we would need to either know the system we have setup or know we can get support)

**Comment by olivierlambert** [comment]
1. If you have an issue form the sources, you can ask for help on the community forum
2. If you pay Citrix only for CBT, I find it weird. Are you aware about XCP-ng? (https://xcp-ng.org). There's no feature restriction whatsoever, only support based model. You can have both XCP-ng+XO entirely for free (no support) or get support on both if you like. I think it will be far easier and cheaper than developing something yourself.

**Comment by mauzilla** [comment]
2) Let me rephrase, we paid Citrix for Standard licenses, but have upgraded to Enterprise Licenses as Standard does not offer CBT, only Enterprise licensing. I looked at XCP-ng, but again, our biggest issue is support, although the product is free, if we ever need support the cost is $600 per year (or around $50 per month per server) which is more than what we would pay for Citrix enterprise licensing.

I really want to support XO, but as we're a small company we're happy to pay but not everyone internally is at a very "pro" level so when something goes down we need some kind of support plan. Maybe we should take the conversation offline and see if there is a middle ground?

**Comment by olivierlambert** [comment]
Feel free to contact us on our live chat on https://xcp-ng.com or https://xen-orchestra.com :)
